python/pyspark/sql/tests.py

```diff
@@ -2367,6 +2380,157 @@ def range_frame_match():
 
         importlib.reload(window)
 
+
+class TypesTest(unittest.TestCase):
+
+    def test_verify_type_ok_nullable(self):
+        for obj, data_type in [
+                (None, IntegerType()),
+                (None, FloatType()),
+                (None, StringType()),
+                (None, StructType([]))]:
```

Comment:
Let's do this like ...

```python
types = [IntegerType()), FloatType()), StringType()), StructType([])]
for ...
```

if you don't mind. I think taking out the same value in a loop is slightly better.